app/controllers/flags_controller.rb

```diff
@@ -0,0 +1,36 @@
+class FlagsController < ApplicationController
+  def create
+    @topic = Topic.find(params[:topic_id])
+    @posts = @topic.posts
+
+    @forum = Forum.isprivate.first
+
+    @topics = @forum.topics.new(
+    	name: "Flagged for review: #{@topic.name}",
+    	private: true)
+
+    @user = current_user
+    @topics.user_id = @user.id
+
+    if @topics.save
+      @flag = Flag.new(
```

Comment:
I am thinking this logic might be reversed, so it only generates the topic if the flag is saved.  Then you could actually move the topic generation into the Flag model- making the code more modular and the controller thinner.  Thoughts?